src/test/kotlin/org/taktik/icure/asynclogic/impl/filter/code/CodeIdsByTypeCodeVersionIntervalFilterTest.kt

```diff
@@ -0,0 +1,129 @@
+package org.taktik.icure.asynclogic.impl.filter.code
+
+import kotlin.random.Random
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.fasterxml.jackson.module.kotlin.SingletonSupport
+import kotlinx.coroutines.flow.fold
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.taktik.icure.asynclogic.CodeLogic
+import org.taktik.icure.asynclogic.impl.filter.Filters
+import org.taktik.icure.services.external.rest.v1.dto.filter.code.CodeIdsByTypeCodeVersionIntervalFilter
+import org.taktik.icure.services.external.rest.v1.mapper.base.CodeMapper
+import org.taktik.icure.test.CodeBatchGenerator
+import org.taktik.icure.test.ICureTestApplication
+import org.taktik.icure.test.removeEntities
+
+@SpringBootTest(
+	classes = [ICureTestApplication::class],
+	properties = ["spring.main.allow-bean-definition-overriding=true"],
+	webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ActiveProfiles("app")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CodeIdsByTypeCodeVersionIntervalFilterTest @Autowired constructor(
+	private val filters: Filters,
+	private val codeLogic: CodeLogic,
+	private val codeMapper: CodeMapper
+) {
+
+	private val testBatchSize = 1001
+	private val codeGenerator = CodeBatchGenerator()
+	private val testBatch = codeGenerator.createBatchOfUniqueCodes(testBatchSize).associateBy { it.id }
+	private val testBatchIds = testBatch.keys.toSortedSet().toList()
+
+	@BeforeAll
+	fun importTestCodeBatch() {
+		runBlocking {
+			testBatch.values.forEach {
+				codeLogic.create(codeMapper.map(it))
+			}
+		}
+	}
+
+	@Test
+	fun allTheIdsAreReturnedIfBothKeysAreNull() {
+		runBlocking {
+			val intervalFilter = CodeIdsByTypeCodeVersionIntervalFilter()
+			val idsCount = filters.resolve(intervalFilter)
+				.fold(0) { acc, it ->
+					assert(testBatchIds.contains(it))
+					acc + 1
+				}
+			Assertions.assertEquals(testBatchIds.size, idsCount)
+		}
+	}
+
+	@Test
+	fun ifStartKeyIsSpecifiedOnlyResultsThatComeAfterAreReturned() {
+		runBlocking {
+			val startIndex = Random.nextInt(0, testBatchIds.size)
+			val startCode = testBatch[testBatchIds[startIndex]]!!
+			val intervalFilter = CodeIdsByTypeCodeVersionIntervalFilter(null, startCode.type, startCode.code, startCode.version, null, null, null)
```

Comment:
As you provided a default value for the properties of the CodeIdsByTypeCodeVersionIntervalFilter, you may create it this way : `CodeIdsByTypeCodeVersionIntervalFilter(startType = startCode.type, startCode = startCode.code, startVersion = startCode.version)` which is more convenient than passing nulls for every one of them

Reply:
Updated in the new commit